planner/core/rule_aggregation_elimination.go

```diff
@@ -100,6 +101,16 @@ func (a *aggregationEliminateChecker) tryToEliminateDistinct(agg *LogicalAggrega
 				}
 				if distinctByUniqueKey {
 					af.HasDistinct = false
+					if opt.tracer != nil {
+						opt.tracer.AppendRuleTracerStepToCurrent(
+							tracing.LogicalRuleOptimizeTraceStep{
```

Comment:
Do not put trace details in rule logic, move them to a tracer func, just pass planNode/expression... Each struct/function should have clean responsibility.

Reply:
updated.